notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,512 @@
+# - Using the true variable believed to be confounded (i.e. method 3 where only travel time and cost are used to recover the confounder) leads to a better recovery of the true confounder. This suggests that it may be better to run the deconfounder algorithm based on a hypothesized causal graph, rather than just running it on all the observed covariates. 
+# - Similar to what we found in the investigating_decounfounder notebook, the effectiveness of the deconfounder algorithm is very sensitive to small deviations in the recovered confounder. Although method 3 returns a relatively good fit of the true confounder, the adjusted coefficients on travel time and cost do not improve, and the coefficients on the recovered confounder are highly insignificant. This raises questions about the usefulness of the deconfounder algorithm in practice.
+
+# ## Useful function
+
+def specifications(mnl_specification, num_modes):
+    newDict= dict()
+
+    for i in range(1,num_modes+1):
+        variables = []
+        # Iterate over all the items in dictionary and filter items which has even keys
+        for (key, value) in mnl_specification.items():
+           # Check if key is even then add pair to new dictionary
+    
+
+
+
+            if any(isinstance(sub, list) for sub in value):
+                for sub in value:
+                    if isinstance(sub, list):
+                        if (i in sub): 
+                            variables.append(key)
+                            
+                    else:
+                        if i == sub:
+                            variables.append(key)
+
+            else:
+                if i in value:
+        #             print(variables)
+
+                    variables.append(key)
+
+        newDict[i] = variables
+
+    return newDict
+
+# ## Analysis
+
+PATH = '../../data/raw/'
+file_name = 'simulated_long_format_bike_data.csv'
+
+data = pd.read_csv(PATH+file_name)
+data = data.drop('Unnamed: 0', axis = 1)
+data.columns
+
+# +
+X_columns = ['total_travel_time',
+       'total_travel_cost', 'total_travel_distance', 
+             'cross_bay', 'household_size', 'num_kids', 
+              'cars_per_licensed_drivers', 
+             'gender'
+             
+            ]
+
+y_column = data['mode_id']
+
+# +
+## Specify the true causal graph used in the simulation of the choices
+
+causal_graph = CausalGraphicalModel(
+    nodes=["Travel_Time", "Travel_Distance", "Travel_Cost", "Cross_Bay_Bridge", "HH_Size", "num_of_kids_household",
+          "Autos_per_licensed_drivers", "Gender", "Mode_Choice"],
+    edges=[
+        ("Travel_Time", "Mode_Choice"), 
+        ("Travel_Distance", "Mode_Choice"), 
+        ("Travel_Cost", "Mode_Choice"), 
+        ("Cross_Bay_Bridge", "Mode_Choice"), 
+        ("HH_Size", "Mode_Choice"), 
+        ("num_of_kids_household", "Mode_Choice"), 
+        ("Autos_per_licensed_drivers", "Mode_Choice"), 
+        ("Gender", "Mode_Choice"), 
+        
+        
```

Comment:
nit: I would remove the additional blank lines. I don't think this is a PEP8 issue, just personal preference. Feel free to ignore if you feel strongly!